### ISSUE TITLE
[BUG] SearchResponseTests.testFromXContent failure

### ISSUE THREAD

**Issue by andrross** [opensearch-project/OpenSearch#5810]
The following test failure is reproducible:

```
./gradlew ':server:test' --tests "org.opensearch.action.search.SearchResponseTests.testFromXContent" -Dtests.seed=1BEA0CB973664D92 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=da-DK -Dtests.timezone=Cuba -Druntime.java=17
```

```
JacksonYAMLParseException: The incoming YAML document exceeds the limit: 3145728 code points.
```

**Comment by andrross** [comment]
A similar failure (different test) was fixed in #5285

**Comment by andrross** [comment]
Just confirmed this test no longer fails with the given seed on either the `main` branch or `2.x`. Closing.
